core/models/department.py

```diff
@@ -0,0 +1,26 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class Department(models.Model):
+    """ Stores all departments """
+
+    name = models.CharField(unique=True, max_length=50)
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
+    last_modified = models.DateTimeField(auto_now=True, editable=False)
+
+    def clean(self):
+        self.name = " ".join(self.name.title().split())
+
+    def save(self, *args, **kwargs):
+        try:
+            self.full_clean()
+        except Exception as e:
+            raise ValidationError(e)
```

Comment:
Raise a DRF exception here

<img width="946" alt="screen shot 2018-08-09 at 12 00 21" src="https://user-images.githubusercontent.com/10899666/43889179-1b21d956-9bcc-11e8-8131-7e5a7103c33a.png">

Reply:
@esirK I could be wrong as to whether you need to raise the DRF exception in the model. Let's close this soonest, so that we can unblock the subsequent tasks that depend on these changes.

Reply:
Ah! this solves it.
```python
from rest_framework import serializers
...

except ValidationError as error:
            raise serializers.ValidationError(error)
```

Reply:
So no changes need to be implemented right?? @collinmutembei